packages/components/src/internal/components/chart/TrendlineOption.tsx

```diff
@@ -17,18 +27,25 @@ interface TrendlineOptionProps {
 export const TrendlineOption: FC<TrendlineOptionProps> = memo(props => {
     const TRENDLINE_OPTIONS: TrendlineType[] = Object.values(LABKEY_VIS.GenericChartHelper.TRENDLINE_OPTIONS);
     const { fieldValues, onFieldChange, schemaQuery } = props;
+    const showFieldOptions = fieldValues.trendlineType?.showMin || fieldValues.trendlineType?.showMax;
 
     // hide the trendline option if no x-axis value selected and for date field selection on x-axis
     const hidden = useMemo(() => {
-        const jsonType = fieldValues.x?.data?.jsonType || fieldValues.x?.data?.type;
+        const jsonType = getFieldDataType(fieldValues.x?.data);
         return !fieldValues.x?.value || jsonType === 'date' || jsonType === 'time';
     }, [fieldValues.x]);
 
     const [loadingTrendlineOptions, setLoadingTrendlineOptions] = useState<boolean>(true);
+    const [asymptoteType, setAsymptoteType] = useState<string>('automatic');
     const [asymptoteMin, setAsymptoteMin] = useState<string>('');
     const [asymptoteMax, setAsymptoteMax] = useState<string>('');
+    const invalidRange = useMemo(
+        () => !!asymptoteMin && !!asymptoteMax && asymptoteMax <= asymptoteMin,
+        [asymptoteMin, asymptoteMax]
+    );
     useEffect(() => {
         if (loadingTrendlineOptions && (!!fieldValues.trendlineAsymptoteMin || !!fieldValues.trendlineAsymptoteMax)) {
+            setAsymptoteType('manual');
```

Comment:
Below this line, on line 55 and 59 we have some `any` types that can be changed to `event: ChangeEvent<HTMLInputElement>`. They're from a previous PR but we should update them in this PR. Tagging this line because I can't tag untouched lines.

Reply:
done